src/main/cpp/xmlhw/CanCoderDefn.cpp

```diff
@@ -0,0 +1,56 @@
+#include <memory>
+#include <string>
+#include <xmlhw/CanCoderDefn.h>
+#include <pugixml/pugixml.hpp>
+#include <utils/Logger.h>
+#include <ctre/Phoenix.h>
+#include <ctre/phoenix/sensors/CANCoder.h>
+
+using namespace std;
+using namespace frc;
+using namespace pugi;
+
+std::shared_ptr<ctre::phoenix::sensors::CANCoder> CanCoderDefn::ParseXML
+(
+    pugi::xml_node CanCoderNode
+)
+{
+    shared_ptr<ctre::phoenix::sensors::CANCoder> cancoder = nullptr;
+
+    string usage;
+    int canID = 0;
+    
+    bool hasError = false;
+
+    for(pugi::xml_attribute attr = CanCoderNode.first_attribute(); attr &&!hasError; attr = attr.next_attribute() )
+    {
+        if ( strcmp( attr.name(), "usage") == 0)
+        {
+            usage = attr.value();
+        }
+        else if ( strcmp(attr.name(),"canID" ) == 0 )
+        {
+            int iVal = attr.as_int();
+            if (iVal > -1 && iVal < 63)
+            {
+                canID = attr.as_int();
+            }
```

Comment:
there is a utility method for this  see motordefn

        else if ( strcmp( attr.name(), "canId" ) == 0 )
        {
            canID = attr.as_int();
            hasError = HardwareIDValidation::ValidateCANID( canID, string( "MotorDefn::ParseXML" ) );
        }